gcc8.2 unsigned int to unsigned char conversion warning

gcc 8.2 & cmake 3.12

Compiler output:
> In file included from ../src/lib_json/../../include/json/writer.h:10,
>                  from ../src/lib_json/json_writer.cpp:8:
> ../src/lib_json/../../include/json/value.h: In member function ‘void Json::Value::setType(Json::ValueType)’:
> ../src/lib_json/../../include/json/value.h:615:51: warning: conversion from ‘unsigned int’ to ‘unsigned char’ may change value [-Wconversion]
>    void setType(ValueType v) { bits_.value_type_ = v; }

--- a/include/json/value.h
+++ b/include/json/value.h
@@ -610,7 +610,7 @@ Json::Value obj_value(Json::objectValue); // {}
   ptrdiff_t getOffsetLimit() const;
 
 private:
-  void setType(ValueType v) { bits_.value_type_ = v; }
+  void setType(ValueType v) { bits_.value_type_ = static_cast<unsigned char> (v); }
   bool isAllocated() const { return bits_.allocated_; }
   void setIsAllocated(bool v) { bits_.allocated_ = v; }
 

